torch/csrc/jit/type.cpp

```diff
@@ -248,10 +254,20 @@ TypePtr matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env) {
   } else if (auto opt_formal = formal->cast<OptionalType>()) {
     if (auto opt_actual = actual->cast<OptionalType>()) {
       return OptionalType::create(matchTypeVariables(
-          opt_formal->getElementType(), opt_actual->getElementType(), type_env));
+          opt_formal->getElementType(),
+          opt_actual->getElementType(),
+          type_env));
     } else {
+      if (actual->isSubtypeOf(NoneType::get())) {
+        return OptionalType::create(opt_formal->getElementType());
```

Comment:
How would this work if opt_formal ElementType is VarType?

Reply:
Doing something like `torch.jit._unwrap_optional(None)` throws an error since `t` doesn't get bound to anything

Reply:
Since None and Int both subtype Optional[int] I think that should work...